client/src/components/JobInfoStack.tsx

```diff
@@ -0,0 +1,54 @@
+import { formatDate, formatSalary, Job } from '../lib';
+import { Link } from 'react-router-dom';
+
+type InfoStackProps = {
+  job: Job;
+};
+
+export default function JobInfoStack({ job }: InfoStackProps) {
+  const {
+    title,
+    companyName,
+    companyImgUrl,
+    location,
+    level,
+    dateAdded,
+    salaryFloor,
+    salaryCeiling,
+  } = job;
+  return (
+    <div className="flex flex-col gap-4">
+      <img
+        className="rounded-full w-24 h-24"
+        src={`../${companyImgUrl}`}
+        alt={companyName}
+      />
+      <h1 className="text-2xl">{title}</h1>
+      <div>
+        <p className="text-sm">Company</p>
+        <p className="text-lg">{companyName}</p>
+      </div>
+      <div>
+        <p className="text-sm">Location</p>
+        <p className="text-lg">{location}</p>
+      </div>
+      <div>
+        <p className="text-sm">Level</p>
+        <p className="text-lg">{level}</p>
+      </div>
+      <div>
+        <p className="text-sm">Salary</p>
+        <p className="text-lg">{formatSalary(salaryFloor, salaryCeiling)}</p>
+      </div>
+      <div>
+        <p className="text-sm">Date Posted</p>
+        <p className="text-lg">{formatDate(dateAdded)}</p>
+      </div>
+      <Link to={'/'}>
+        <button className="text-sm py-1 px-3 border-[1.5px] border-stone-700 rounded hover:bg-stone-300 transition-colors">
```

Comment:
This is the thing I don't like about Tailwind! Not only is this long and cumbersome, but you probably will have several of these buttons all with exactly the same list of class names, and if you decide to change your look you'll have to find all of them and change them all!

To deal with that, I usually create a set of custom button components so I can just use the component and not have all this code every time. In this case, I'd probably create a ButtonLink (or LinkButton) that takes `to` and `text` as props and renders both the Link and button components.

Reply:
Agreed, that has been annoying to deal with.

I was planning on making some button components, but I didn't think to put the Link in that component. I will definitely do that!